tls/s2n_tls_parameters.h

```diff
@@ -194,9 +194,15 @@
  */
 #define S2N_TLS_RECORD_HEADER_LENGTH    5
 #define S2N_TLS_MAXIMUM_FRAGMENT_LENGTH 16384
-#define S2N_TLS_MAXIMUM_RECORD_LENGTH   (S2N_TLS_MAXIMUM_FRAGMENT_LENGTH + S2N_TLS_RECORD_HEADER_LENGTH)
+/* Maximum TLS record length allows for 2048 octets of compression expansion and padding */
+#define S2N_TLS_MAXIMUM_RECORD_LENGTH   (S2N_TLS_MAXIMUM_FRAGMENT_LENGTH + S2N_TLS_RECORD_HEADER_LENGTH + 2048)
```

Comment:
hmm, i would consider S2N_TLS_MAXIMUM_RECORD_LENGTH to be without S2N_TLS_RECORD_HEADER_LENGTH, but the math works with regards to the way s2n use this variable for the message/buffer. we just have to make sure elsewhere takes into account this header length when this constant.

Reply:
Makes sense.